src/lib.rs

```diff
@@ -348,6 +348,304 @@ languages_table!(
     (Zu, "zu", "zul", "zul", "Zulu", "Niger–Congo"),
 );
 
+#[cfg(feature = "iterable")]
+pub mod iter {
+    use super::LanguageCode;
+
+    #[derive(Debug, Default, Clone)]
+    pub struct Iter(usize);
+
+    impl Iterator for Iter {
+        type Item = &'static LanguageCode;
```

Comment:
It would be better to return an owned `LanguageCode` instead of a static reference, because then the list of `LanguageCodes` would not have to be included in the resulting binary.

Reply:
Ah I missed this comment...
Not sure I understand how it can prevent including the array in the resulting binary tho.

Reply:
Well I think I grasp the idea maybe, you mean something like :
```rust
impl Iterator for Iter {
    type Item = LanguageCode;

    fn next(&mut self) -> Option<Self::Item> {
        if self.0 == 184 {
            return None;
        }
        let language = match self.0 {
            0 => LanguageCode::Ab,
            1 => LanguageCode::Aa,
            2 => LanguageCode::Af,
            3 => LanguageCode::Ak,
            // ...
            183 => LanguageCode::Zu,
            _ => unreachable!(),
        };
        self.0 += 1;
        Some(language)
    }
}
```
?
Then shouldn't it be turned into a `into_iter` instead of `iter`, since it will yield `T` instead of `&T` ?

Reply:
If LLVM sees that you're indexing into an array of increasing numbers and just returning the value, it will likely optimize it out and return that number directly.

Reply:
> So you don't have to use a match.
You mean that I should switch `match` for something else ?

Reply:
You would keep all existing code (with the array index and without a very long match expression), and only change it so that `LanguageCode` is used instead of `&'static LanguageCode`.

Reply:
Ok done I guess